### DISCUSSION TITLE
Are static assets cached between requests?

### DISCUSSION THREAD

**Discussion by colinmollenhour** [uyamazak/hc-pdf-server#651] [Q&A] [answered]
I don't really know how Puppeteer works or how this wrapper manages it so maybe it is a simple answer. If I have multiple requests which need to load static assets (e.g. images, scripts) and assuming those are cacheable will they be cached or will they be fetched fresh for each request?

**Comment by uyamazak** [comment, marked as answer]
@colinmollenhour 
Hi, 

I have not checked the detailed behavior, it may be `cacheable will they be cached`, per Puppeteer's Page. 

Because Puppeteer's Page has setCacheEnabled() method defaults to `true`.

https://pptr.dev/api/puppeteer.page.setcacheenabled

Note that hc-pdf-server uses multiple Pages by default.

If you want to Page.setCacheEnabled() to false, I will consider adding the feature to switch in env.

**Reply by colinmollenhour** [reply]
Thanks for the info! Caching enabled is actually the behavior I want so no need to add an option for it on my behalf.
